rdr_service/dao/bq_questionnaire_dao.py

```diff
@@ -113,9 +113,20 @@ def make_bqrecord(self, p_id, module_id, latest=False, convert_to_enum=False):
             responses = session.execute(_participant_module_responses_sql, {'module_id': module_id, 'p_id': p_id,
                                                                             'test_hpo': TEST_HPO_NAME,
                                                                             'system': PPI_SYSTEM})
+            answer_hashes_processed = []
             for qr in responses:
+                if qr.answer_hash in answer_hashes_processed:
+                    # PDR-863:  This response is pending being flagged by RDR as a duplicate (same answer hash as
+                    # a more recently created response).  Proactively exclude from PDR data to avoid orphaned records
+                    continue
+
+                answer_hashes_processed.append(qr.answer_hash)
```

Comment:
Does this only compare answer hashes? We could receive some responses multiple times that have the same answer hash (such as responding multiple times to EHR or GROR consent). Using external id as part of the comparison could help make sure distinct responses that have the same answers are accounted for.

Reply:
I'm not sure how often it happens, but there are some surveys that the meta-data and timing relative to other responses matters as much (if not more) than the answer content. Such as if a participant goes back and forth on providing Yes and No responses to EHR/GROR. Or responses to digital health sharing data that don't have answers, but instead depend on the survey code being used (like `participantintendstosharefitbit` and `participantintendstostopsharingfitbit`).

Will PDR users only ever need the the latest response for a given survey, or is it possible that they'd need time ranges for things like the consents above?

Reply:
I think the authored timestamp is the more important differentiator.  E.g, if it's a different authored time but everything else is the same then we should keep both.   if it's an identical authored time and answered hash, we just want the most recently received "copy".

I'll put this on hold to revise.

Reply:
Sounds good. I agree that authored time would be ideal, and I think we would have gone with that for the duplicate detection in RDR. But we've seen that the authored times can shift by a small amount between replays of responses, so for something we know to be a replay of a response we've seen differing authored times but the same external id.